src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -603,9 +610,9 @@ - (void)updateWithDelegate:(chip::DeviceLayer::BleScannerDelegate *)delegate
                 _scannerDelegate->OnBleScanAdd((__bridge void *) cachedPeripheral, info);
             });
         }
+    } else {
+        [self setupTimer:kScanningWithoutDelegateTimeoutInSeconds];
```

Comment:
OK, so this looks like it can call `setupTimer` while `_timer` is not nil, if the old mode was also `kScanningWithoutDelegate`?

Reply:
I have added additional clean up code at the beginning of `setupTimer`.